src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -651,32 +649,29 @@ internal static void UnregisterMemoryLoadChangeNotification(Action notification)
             }
         }
 
-        // Skips zero-initialization of the array if possible. If T contains object references,
-        // the array is always zero-initialized.
-        internal static T[] AllocateUninitializedArray<T>(int length)
+        /// <summary>
+        /// Skips zero-initialization of the array if possible. If T contains object references,
+        /// the array is always zero-initialized.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] // forced to ensure no perf drop for small memory buffers (hot path)
+        internal static T[] AllocateUninitializedArray<T>(uint length)
         {
             if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
             {
                 return new T[length];
             }
 
-            if (length < 0)
-                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.lengths, 0, ExceptionResource.ArgumentOutOfRange_NeedNonNegNum);
-#if DEBUG
```

Comment:
Why did you remove `#if DEBUG`? It was here to improve the code coverage and I believe that it was proven to be useful to find number of bugs earlier.